src/licenses.ts

```diff
@@ -87,15 +87,19 @@ export async function getInvalidLicenseChanges(
     } else if (validityCache.get(license) === undefined) {
       try {
         if (allow !== undefined) {
-          const found = allow.find(spdxExpression =>
-            spdxSatisfies(license, spdxExpression)
-          )
-          validityCache.set(license, found !== undefined)
+          if (spdx.isValid(license)) {
+            const found = spdx.satisfiesAny(license, allow)
+            validityCache.set(license, found)
+          } else {
+            invalidLicenseChanges.unresolved.push(change)
+          }
         } else if (deny !== undefined) {
-          const found = deny.find(spdxExpression =>
-            spdxSatisfies(license, spdxExpression)
-          )
-          validityCache.set(license, found === undefined)
+          if (spdx.isValid(license)) {
+            const found = spdx.satisfiesAny(license, deny)
+            validityCache.set(license, !found)
+          } else {
+            invalidLicenseChanges.unresolved.push(change)
+          }
         }
```

Comment:
⚠️ one more note about the new uses of `satisfiesAny` in these evaluations. This should correctly resolve the problems listed in the support Issues that this PR will close out ...except for [this one](https://github.com/actions/dependency-review-action/issues/263). 

As the new test suites for SPDX expression eval show, we could use `satisfies` in place of `satisfiesAny` to make not only the incoming package license spec parse as an expression (as this PR enables) but also treat our allow/deny list entries as expressions. This would enable DR Action config to set not only single-license entries, but expressions like `MIT AND Apache-2.0` and so on.

But, to resolve the case described in the linked support Issue (incoming package includes an `AND` expression) the end user would still need to supply some duplicative entries in the expression-based allow/deny lists to ensure a good match...

Seems like closing out that last Issue should happen after this PR, and after mulling over some product-facing options for supporting these expressions better 👍

Reply:
Thanks @elireisman, the description for this PR indicated that it closes that issue but I was able to confirm that it does not. I removed that line from the description :(